src/manifest.json.mustache

```diff
@@ -40,53 +46,100 @@
   {{/browserIsChrome}}
 
   "options_ui": {
-    "page": "options/index.html",
-    "chrome_style": true
+    "page": "options/index.html"
   },
 
   {{#browserIsChrome}}
   "offline_enabled": false,
   {{/browserIsChrome}}
 
   "permissions": [
+    {{#manifestV3}}
+    "scripting",
+    {{/manifestV3}}
+
+    {{^manifestV3}}
     "<all_urls>",
+    {{/manifestV3}}
 
     "storage",
     "tabs"
   ],
 
+  {{#manifestV3}}
+  "host_permissions": ["<all_urls>"],
+  {{/manifestV3}}
+
   "optional_permissions": [
     {{! Used to enumerate frames on certain websites. }}
     "webNavigation"
   ],
 
+  {{^manifestV3}}
   "content_security_policy": "script-src 'self'; object-src 'self'",
+  {{/manifestV3}}
 
+  {{#manifestV3}}
+  "background": {
+    "service_worker": "extension.bundle.js"
+  },
+  {{/manifestV3}}
+
+  {{^manifestV3}}
   "background": {
     {{#browserIsChrome}}
     "persistent": true,
     {{/browserIsChrome}}
     "scripts": [
       "extension.bundle.js"
-    ]
+     ]
   },
+  {{/manifestV3}}
+
+  {{#manifestV3}}
+  "action": {
+    "default_icon": {
+      "19": "images/browser-icon-inactive.png",
+      "38": "images/browser-icon-inactive@2x.png"
+    }
+  },
+  {{/manifestV3}}
+
+  {{^manifestV3}}
   "browser_action": {
     "default_icon": {
       "19": "images/browser-icon-inactive.png",
       "38": "images/browser-icon-inactive@2x.png"
     }
   },
+  {{/manifestV3}}
```

Comment:
Suggestion: maybe we could do this:

```
{{#manifestV3}}
"action": {
{{/manifestV3}}
{{^manifestV3}}
"browser_action": {
{{/manifestV3}}
  "default_icon": {
    "19": "images/browser-icon-inactive.png",
    "38": "images/browser-icon-inactive@2x.png"
  }
},
```

That way we define the icons only once and it's less likely to forget editing one.

On the other hand we probably get rid of most of this logic soon, so maybe it's not worth it.